blockly-core/core/ui/contract_editor/domain_editor.js

```diff
@@ -11,58 +11,58 @@ goog.require('Blockly.XButton');
  * A DOM-based parameter editor
  * @param {!Object} options
  * @param {string} options.name
+ * @param {string} options.paramID unique ID of domain
  * @param {Blockly.BlockValueType} options.type
  * @param {Function} options.onRemovePress
  * @param {Function} options.onTypeChanged takes {Blockly.BlockValueType} newType
- * @param {Function} options.onNameChanged takes {string} newName
+ * @param {Function} options.onNameChanged takes {string} uniqueID
  * @param {Object.<String, Blockly.BlockValueType>} options.typeChoices
  * @constructor
  */
 Blockly.DomainEditor = function (options) {
-  this.onRemovePress = options.onRemovePress;
-  this.onTypeChanged = options.onTypeChanged;
-  this.onNameChanged = options.onNameChanged;
-  this.typeChoices = options.typeChoices;
-  this.type = options.type;
-  this.name = options.name;
+  this.options = options;
```

Comment:
I decided to leave these as an `options` object to be a bit more explicit about what they are / do (i.e., just initial options, not necessarily updated as state changes).